contracts/vault/src/query.rs

```diff
@@ -0,0 +1,48 @@
+use perpswap::contracts::vault::QueryMsg;
+
+use crate::{common::get_total_assets, prelude::*, state};
+#[allow(dead_code)]
+/// Handles all queries to the vault
+///
+/// # Parameters
+/// - `deps`: Dependencies for storage access
+/// - `env`: Contract environment
+/// - `msg`: Query message
+///
+/// # Returns
+/// - `StdResult<Binary>`: Serialized query response
+#[entry_point]
+pub fn query(deps: Deps, env: Env, msg: QueryMsg) -> StdResult<Binary> {
+    match msg {
+        QueryMsg::GetVaultBalance {} => {
+            let config = state::CONFIG.load(deps.storage)?; // Load configuration
+            let balance = deps
+                .querier
+                .query_balance(&env.contract.address, &config.usdc_denom)?
+                .amount;
+            to_json_binary(&balance) // Return vault balance
+        }
+        QueryMsg::GetPendingWithdrawal { user } => {
+            let pending = state::PENDING_WITHDRAWALS
+                .may_load(deps.storage, &user)?
+                .unwrap_or(Uint128::zero());
+            to_json_binary(&pending) // Return user's pending withdrawal
```

Comment:
Generally speaking, I recommend that all query responses be JSON objects. It allows us to expand the responses with more data without breaking API compatibility in the future.